File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    xcnt, cnt = 0, 0
    x = s[0]
    for i in range(len(s)):
        if x == s[i]:
            xcnt += 1
        else:
            cnt += 1
        if xcnt == cnt:
            answer += 1
            xcnt, cnt = 0, 0
            if i != len(s) - 1:
                x = s[i+1]
    if cnt != xcnt:
        answer += 1

    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0;
    let cnt = 0;
    let cur = '';

    for (let i=0; i<s.length; i++) {
        if (cnt === 0) {
            cnt = 1;
            answer++;
            cur = s[i];
        } else {
            if (cur === s[i]) cnt++;
            else cnt--;
        }
    }
    return answer;
}